eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        Assert.assertNull(EurekaEntityFunctions.selectInstance(
+                new Applications(), "foo"));
+        Assert.assertNull(EurekaEntityFunctions.selectInstance(
+                new Applications(), "foo", "foo"));
+
+        Assert.assertEquals(instanceInfo, EurekaEntityFunctions.selectInstance(
+                applications, "foo"));
+        Assert.assertEquals(instanceInfo, EurekaEntityFunctions.selectInstance(
+                applications, "foo", "foo"));
+    }
+
+    @Test
+    public void testTakeFirstIfNotNullReturnFirstInstance() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        Assert.assertNull(EurekaEntityFunctions.takeFirst(new Applications()));
+
+        Assert.assertEquals(instanceInfo,
+                EurekaEntityFunctions.takeFirst(applications));
+    }
+
+    @Test
+    public void testSelectAllIfNotNullReturnAllInstances() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        ArrayList<InstanceInfo> retval = new ArrayList<>();
+        retval.add(instanceInfo);
+        retval.add(instanceInfo);
+        Assert.assertEquals(retval,
+                EurekaEntityFunctions.selectAll(applications));
+    }
+
+    @Test
+    public void testToApplicationMapIfNotNullReturnMapOfApplication() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getAppName()).thenReturn("foo");
+
+        ArrayList<InstanceInfo> instanceInfos = new ArrayList<>();
+        instanceInfos.add(instanceInfo);
+
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertEquals(1,
```

Comment:
Add another assert checking the map does contain an application objected keyed by the expected name.

Reply:
Done